src/markov/markov_approx.jl

```diff
@@ -219,3 +223,548 @@ function estimate_mc_discrete{T}(X::Vector{T})
+  Quantitative Economics, 8(2), 651-683.
+
+"""
+function discrete_var(b::Union{Real, AbstractVector},
+                      B::Union{Real, AbstractMatrix},
+                      Psi::Union{Real, AbstractMatrix},
+                      Nm::Integer,
+                      n_moments::Integer=2,
+                      method::VAREstimationMethod=Even(),
+                      n_sigmas::Real=sqrt(Nm-1))
+
+    M, M_ = size(B, 1), size(B, 2)
+
+    # Check size restrictions on matrices
+    M == M_ || throw(ArgumentError("B must be a scalar or square matrix"))
+    M == length(b) || throw(ArgumentError("b must have the same number of rows as B"))
+
+    #% Check that Psi is a valid covariance matrix
+    isposdef(Psi) || throw(ArgumentError("Psi must be a positive definite matrix"))
+
+    # Check that Nm is a valid number of grid points
+    Nm >= 3 || throw(ArgumentError("Nm must be a positive interger greater than 3"))
+
+    # Check that n_moments is a valid number
+    if n_moments < 1 || !(n_moments % 2 == 0 || n_moments == 1)
+        error("n_moments must be either 1 or a positive even integer")
+    end
+
+    # Compute polynomial moments of standard normal distribution
+    gaussian_moment = zeros(n_moments)
+    c = 1
+    for k=1:floor(Int, n_moments/2)
+        c = (2*k-1)*c
+        gaussian_moment[2*k] = c
+    end
+    # Compute standardized VAR(1) representation
+    # (zero mean and diagonal covariance matrix)
+    A, C, mu, Sigma = standardize_var(b, B, Psi, M)
+
+    # Construct 1-D grids
+    y1D = construct_1D_grid(Sigma, Nm, M, n_sigmas, method)
+
+    # Construct all possible combinations of elements of the 1-D grids
+    D = allcomb3(y1D')'
+
+    # Construct finite-state Markov chain approximation
+    # conditional mean of the VAR process at each grid point
+    cond_mean = A*D
+    # probability transition matrix
+    P = ones(Nm^M, Nm^M)
+    # normalizing constant for maximum entropy computations
+    scaling_factor = y1D[:, end]
+    # used to store some intermediate calculations
+    temp = Array{Float64}(M, Nm)
+    # store optimized values of lambda (2 moments) to improve initial guesses
+    lambda_bar = zeros(2*M, Nm^M)
+    # small positive constant for numerical stability
+    kappa = 1e-8
+
+    for ii = 1:(Nm^M)
+
+        # Construct prior guesses for maximum entropy optimizations
+        q = construct_prior_guess(cond_mean[:, ii], Nm, y1D, method)
+
+        # Make sure all elements of the prior are stricly positive
+        q[q.<kappa] = kappa
+
+        for jj = 1:M
+            # Try to use intelligent initial guesses
+            if ii == 1
+                lambda_guess = zeros(2)
+            else
+                lambda_guess = lambda_bar[(jj-1)*2+1:jj*2, ii-1]
+            end
+
+            # Maximum entropy optimization
+            if n_moments == 1 # match only 1 moment
+                temp[jj, :], _, _ = discrete_approximation(y1D[jj, :],
```

Comment:
Sorry to bring this up now, but is there a reason we _need_ temp to be ordered the way it is? The code might be more efficient if it was laid out in column major order (e.g. using `temp[:, jj]` instead of `temp[jj, :]`). 

I don't think this will make a big difference, especially relative to the cost of calling solvers, so if you have a reason for the current ordering I won't push too hard on this.